### ISSUE TITLE
v0.20.0/0.21.0 makes data after a nullable column null

### ISSUE THREAD

**Issue by Linksku** [MaterializeInc/materialize#10981]
### What version of Materialize are you using?

v0.20.0/v0.21.0

### How did you install Materialize?

APT package

### What was the issue?

I've been running Materialize for several months without issues. I'm using a direct Postgres source. After upgrading to 0.20.0, if there's null data, all subsequent columns in that row become null too.

E.g. here's the data in the source Postgres table:
![Capture](https://user-images.githubusercontent.com/7908179/156344266-0c1786e1-347b-49f8-8499-03babefb3290.PNG)

In 0.19.0 and earlier, the view in Materialize would look exactly the same. In v0.20.0, it becomes:
![Capture2](https://user-images.githubusercontent.com/7908179/156344413-1deff663-e32a-4cc6-a6a0-204718ac2d0e.PNG)

I tried reinstalling Materialize and Postgres several times and deleting the data folders. There's no errors in the error log. After downgrading to 0.19.0, it works again. I don't think I have any unusual Postgres settings.

The view was created according to the Postgres CDC guide:

```
Postgres:
ALTER TABLE __test__ REPLICA IDENTITY FULL');
CREATE PUBLICATION mz_source_test FOR TABLE __test__

Materialize:
CREATE MATERIALIZED SOURCE mz_source_test
FROM POSTGRES
CONNECTION 'host=127.0.0.1 port=5432 user= password= dbname= sslmode=require'
PUBLICATION 'mz_source_test';
CREATE VIEWS FROM SOURCE mz_source_test (__test__);
```

In addition, whenever I update one of the rows containing null in Postgres, then select from Materialize, I get:

```
materialize=> select * from __test__;
ERROR:  Invalid data in source, saw retractions (1) for row that does not exist: [Int32(99525), List([String("1"), Null, String("1")]), Int64(1), Null, Int64(1)]
```

Nothing in the release notes seemed related, any ideas?

### Versions

Postgres: 14.2
OS: Ubuntu 20.04 on WSL 2

**Comment by petrosagg** [comment]
hi @Linksku, I'm trying to reproduce this but I'm having trouble. Here are the exact steps I tried:

First, I ran a postgres container with the same version as yours like so:
```bash
docker run -p 5432:5432 -e POSTGRES_PASSWORD=postgres postgres:14.2 -c wal_level=logical
```

Then I connected to postgres and ran the following commands:

```sql
CREATE TABLE __test__ (id int, name text, id2 int);
INSERT INTO __test__ VALUES (1, NULL, 1);
INSERT INTO __test__ VALUES (2, 'name2', 2);
INSERT INTO __test__ VALUES (3, NULL, 3);
INSERT INTO __test__ VALUES (4, 'name4', 4);
ALTER TABLE __test__ REPLICA IDENTITY FULL;
CREATE PUBLICATION mz_source_test FOR TABLE __test__;
```

Then I connected to materialize v0.20.0 and ran the following commands:

```sql
CREATE MATERIALIZED SOURCE mz_source_test FROM POSTGRES CONNECTION 'host=localhost port=5432 user=postgres password=postgres dbname=postgres' PUBLICATION 'mz_source_test';

CREATE VIEWS FROM SOURCE mz_source_test (__test__);
```

After these steps if I query materialize for the `__test__` table I get the correct results:

```
materialize=> CREATE VIEWS FROM SOURCE mz_source_test (__test__);
CREATE VIEW
materialize=> SELECT * FROM __test__;
 id | name  | id2
----+-------+-----
  1 |       |
  3 |       |
  2 | name2 |   2
  4 | name4 |   4
(4 rows)
```

I then added two more rows on the postgres side to ensure that streamed rows are also consumed correctly using these commands:

```sql
INSERT INTO __test__ VALUES (5, NULL, 5);
INSERT INTO __test__ VALUES (6, 'name6', 6);
```

And queries materialize again which still had the correct results:

```
materialize=> SELECT * FROM __test__;
 id | name  | id2
----+-------+-----
  1 |       |
  3 |       |
  5 |       |   5
  2 | name2 |   2
  4 | name4 |   4
  6 | name6 |   6
(6 rows)
```

It would be extremely helpful to see if the above steps work for you and if they don't we need to start figuring out what is special about your other database setup.

**Comment by Linksku** [comment]
It looks like you're getting the same issue as me, shouldn't the correct output be:

```
 id | name  | id2
----+-------+-----
  1 |       |   1
  3 |       |   3
  2 | name2 |   2
  4 | name4 |   4
```

I forgot to mention that newly inserted rows work fine. The issue occurs only with rows that existed before Materialize connects to Postgres.

**Comment by petrosagg** [comment]
LOL. I should get my eyes checked. I'll try to figure out what's going on, thanks!

**Comment by petrosagg** [comment]
@Linksku thanks for reporting this! The fix will be included in the next release which should be out this week if everything goes to plan

**Comment by Linksku** [comment]
Thanks for the quick fix!

**Comment by petrosagg** [comment]
@Linksku the fix is out as part of the `v0.22.0` release. Thanks again for reporting this and let us know if you hit any issues in the future!
